pulsar-client-api/src/main/java/org/apache/pulsar/client/api/PulsarClient.java

```diff
@@ -308,14 +308,32 @@ static ClientBuilder builder() {
      *
      * <p>This can be used to discover the partitions and create {@link Reader}, {@link Consumer} or {@link Producer}
      * instances directly on a particular partition.
-     *
+     * @Deprecated it is not suggested to use now; please use {@link #getPartitionsForTopic(String, boolean)}.
      * @param topic
      *            the topic name
      * @return a future that will yield a list of the topic partitions or {@link PulsarClientException} if there was any
      *         error in the operation.
+     *
      * @since 2.3.0
      */
-    CompletableFuture<List<String>> getPartitionsForTopic(String topic);
+    @Deprecated
+    default CompletableFuture<List<String>> getPartitionsForTopic(String topic) {
+        return getPartitionsForTopic(topic, true);
+    }
+
+    /**
+     * 1. Get the partitions if the topic exists. Return "[{partition-0}, {partition-1}....{partition-n}}]" if a
+     *   partitioned topic exists; return "[{topic}]" if a non-partitioned topic exists.
+     * 2. When {@param metadataAutoCreationEnabled} is "false", neither the partitioned topic nor non-partitioned
+     *   topic does not exist. You will get an {@link PulsarClientException.NotFoundException}.
+     *  2-1. You will get a {@link PulsarClientException.NotSupportedException} if the broker's version is an older
+     *    one that does not support this feature and the Pulsar client is using a binary protocol "serviceUrl".
```

Comment:
`this feature` will be a confusing term. I think the correct description should be

```
You will get a {@link PulsarClientException.NotSupportedException} with metadataAutoCreationEnabled=false on an old broker version which does not support getting partitions without partitioned metadata auto-creation.
```

Reply:
Sure, fixed.